### ISSUE TITLE
False negative examples for axe

### ISSUE THREAD

**Issue by sumleo** [dequelabs/axe-core#1478]
I build some examples which can not be distinguished by the axe 3.2.2 
Expectation:
1 and 2 should be warned by axe.
```html
1.
<ul>
             <li>   <li></li></li>
            </ul>

2.
<ul>
<li>
<div>
             <li></li>
 </div>
 </li>
 </ul>
```

Actual: No warnings.

<pre><code>
axe-core version: 3.2.2
</code></pre>

**Comment by straker** [comment]
Both cases are resolved by the browsers parsing engine and corrected to result in a proper HTML structure.

```html
<ul>
  <li></li>
  <li></li>
</ul>

<ul>
  <li>
    <div></div>
  </li>
  <li></li>
</ul>
```
